### ISSUE TITLE
Intersections between datasets

### ISSUE THREAD

**Issue by laufrant** [UniversalDependencies/docs#227]
Hi,

I ran some tests to check for intersections between train/dev/test datasets and I was surprised by the large numbers I found. Some are duplicates inside train or very short sentences, but there are also many long sentences present in both train and test sets (around 8% of the Spanish test set). More importantly, in many cases the train and the test occurrences have different annotations.

I have uploaded here listings for the duplicates I have found so far in ud-1.1, with X markers for annotation divergences: http://pastebin.com/raw.php?i=fkPU2uzN

I found similar issues in the earlier versions. Had this been already reported?

Thanks for looking into this,
Lauriane

**Comment by spyysalo** [comment]
@laufrant : thank you, that's a very interesting finding, I don't think this has been reported previously. To further summarize your results, I checked the number of overlaps found, excluding short sentences (<10 tokens):

```
bg   32
da   0
de   304
el   1
en   203
es   112
eu   4
fa   2
fi   7
fr   114
ga   0
he   45
hr   2
hu   1
id   6
it   112
sv   2
```

(hope I got that right.)

**Comment by spyysalo** [comment]
Regarding Finnish (only) (@fginter): I checked the Finnish cases of overlap between sentences longer than 10. All but two involved the apparent boilerplate legalese sentence

> Tämä asetus on kaikilta osiltaan velvoittava, ja sitä sovelletaan sellaisenaan kaikissa jäsenvaltioissa.

The other two were

> Tämä asetus tulee voimaan kolmantena päivänä sen jälkeen, kun se on julkaistu Euroopan yhteisöjen virallisessa lehdessä.
> UPM:n mukaan sen tuotteiden kysyntänäkymät vuoden jälkipuoliskolla ovat pääosin vakaat.

I expect these duplications in Finnish represent a real property of the domain text (EU legal documents) and are likely not a cause for concern, although it would be good to cross-check that their annotation is consistent.

**Comment by msimi** [comment]
Do you have a script to check overlappings that I can use?

— Maria

**Comment by spyysalo** [comment]
@msimi : sorry, no, I only had seven cases in Finnish so I just checked the lines in @laufrant's document in a text editor.

@laufrant : could the program you're using to check this be used to show the overlapping sentences?

**Comment by laufrant** [comment]
I changed slightly my output, it should be easier to check the sentences now : http://pastebin.com/raw.php?i=A8BW3Pj1
It is not easy to give you my script, since it is basically a small tweak in my own pipeline. Sorry for that.
Do you need any other output?

At least de/es/fr/it seem to me like true overlaps. For English they are mostly redundancies inside each set, but with annotation divergences though (for instance ADJ becomes PROPN in sentence at lines 7035 and 7627 in train set, change of root for sentence at lines 30629 and 32034…). For the other languages I cannot say.

**Comment by dan-zeman** [comment]
The main issue here has been addressed in UD release 1.2. Thanks for reporting it! It came in the last minute but people did examine their languages and removed overlapping text from dev or test where appropriate (de, es, fi, fr, it). Naturally occurring repetitions were retained.

What may not be solved are inconsistent annotations of duplicate sentences that were retained. I am going to close this issue now. Please open a new issue(s) if inconsistent annotation persists in UD 1.2. (It may actually be valid if there are two different readings of the sentence but I do not think it is too probable.)
